common/commands/src/coconut/generate_freepass.rs

```diff
@@ -55,17 +59,17 @@ pub struct Args {
 
 async fn get_freepass(
     api_clients: Vec<CoconutApiClient>,
-    aggregate_vk: &VerificationKey,
+    aggregate_vk: &VerificationKeyAuth,
     threshold: u64,
     epoch_id: EpochId,
     signing_account: &AccountData,
-    expiration_date: OffsetDateTime,
+    expiration_date_ts: u64,
 ) -> anyhow::Result<IssuedBandwidthCredential> {
-    let issuance_pass = IssuanceBandwidthCredential::new_freepass(Some(expiration_date));
+    let issuance_pass = IssuanceBandwidthCredential::new_freepass(expiration_date_ts);
     let signing_data = issuance_pass.prepare_for_signing();
 
-    let credential_shares = Arc::new(tokio::sync::Mutex::new(Vec::new()));
-
+    let wallet_shares = Arc::new(tokio::sync::Mutex::new(Vec::new()));
+    let signatures_shares = Arc::new(tokio::sync::Mutex::new(Vec::new()));
```

Comment:
it's a tiny thing but imo just keep a single mutex. no point in having two of them since they're always used at the same time anyway

Reply:
Fixed in 0090c6954d5668f225af83f58bd91e7311ce077b